torch/_dynamo/symbolic_convert.py

```diff
@@ -245,12 +245,25 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
                 self.jump(inst)
                 return
 
-            # Manually insert torch._assert instead of python assert and jump over
+            # Manually insert torch._assert_async instead of python assert and jump over
```

Comment:
One interesting question is that, even if we capture the assert, should we also transform it async? If the original assert was an `assert XXX`, the user might be expecting a DtoH sync to happen, and maybe we should respect that, instead of silently turning it into async asserts that can trash the cuda context.

Reply:
Yeah that seems like a problem, but we don't have an `assert` that is blocking, maybe we should add one instead of using assert_async

Reply:
I see. I am not very familiar with CUDA so it will take me sometime to get it right :). I can do it in the follow up diff.